How-To-Use-NDB.md

```diff
@@ -0,0 +1,260 @@
+
+```nim
+for v in cheap_vegs:
+  echo $v[0] & ": " & $v[1]
+```
+
+which prints the remaining items
+
+```
+green bean: 100
+squash: 32
+```
+
+#### READING RECORDS: HANDLING NO RESULTS
+
+If the query has no rows to return (no matches found), then the iterator points to no results. Simply check the iterator for the `finished` condition:
+
+```nim
+import ndb / sqlite
+let db = open("example.db", "", "", "")
+
+var super_expensive = db.rows(sql"SELECT name, qty FROM MyGarden WHERE price > 1000.00")
+
+if finished(super_expensive):
+    echo "nothing super-expensive found"
+else:
+    echo "1 or more super-expensive found"
+```
+
+#### READING RECORDS: GETTING FEWER (OR ONE)
+
+You can set a limit on the results by either passing that limit in the SQL. Or you can add an optional parameter to your `rows` call.
+
+```nim
+import ndb / sqlite
+let db = open("example.db", "", "", "")
+
+var some = db.rows("SELECT name, qty FROM MyGarden LIMIT 2")
+# or you can do:
+some = db.rows("SELECT name, qty FROM MyGarden", 2)
+```
+
+If you are only wanting _one_ result, you can also use the `getRow` function to get the *first* row found. The function returns an `Option[Row]`
+
+```nim
+import ndb / sqlite
+
+let db = open("example.db", "", "", "")
+
+var cucumber = db.rows("SELECT qty FROM MyGarden WHERE name='cucumber")
+
+if cucumber.isNone:
+  echo "No cucumber found."
+else:
+  echo "cucumber: " & $cucumber.get()  # prints "cucumber: 22"
+
+db.close()
+```
+
+**EXTRA:** [Understanding Types](How-To-Use-NDB-understanding-types.md)
+
+## UPDATING RECORDS
+
+Updates are best done using the `execAffectedRows` function. You pass the SQL UPDATE query in a manner similar to the exec function, but this function returns the number of rows that it affected.
+
+```nim
+var count = db.execAffectedRows("""
+    UPDATE MyGarden SET price = 0.90 WHERE name=?
+""", "green bean")
+
+echo $count  # prints "1"
+```
+
+If no rows are updates, you will get a zero (0) returned.
+
+## DELETING RECORDS
+
+Deletions are best done using the `execAffectedRows` function. You pass the SQL DELETE query in a manner similar to the exec function, but this function returns the number of rows that it affected.
+
+```nim
+var count = db.execAffectedRows("""
+    DELETE FROM MyGarden WHERE name=?
+""", "carrot")
+
+echo $count  # prints "1"
+```
+
+If no rows are deleted, you would get a zero (0) returned.
+
+## CATCHING ERRORS
+
+TODO
+
+## [NOTES ABOUT THIS DOC]
+
+This notes section will go away on first release. Some stuff jotted down.
+
+This doc is written with the presumption that the notes marked here are implemented rather than how things work right now.
+
+* Adding a `isNull` function. (and possibly `isNotNull`)
+* We deprecate the `sql` string distinction as it seems to do nothing more than add complexity. However, for legacy purposes, the sql type is not removed.
+* db_common is made part of the library in preparation of movement of `ndb` to the standard library. As such:
```

Comment:
Sounds reasonable.

Also, (if everything goes well) ndb will not become a part of the standard library, it will be referred in docs, and the standard `db_*` modules will be moved to the [graveyard].
https://github.com/nim-lang/Nim/issues/9453#issuecomment-435482881

[graveyard]: https://github.com/nim-lang/graveyard